shadow-dom/leaktests/selection.html

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta name='author' title='weizman' href='https://www.weizmangal.com'>
+<meta name='assert' content='Shadow DOM should not leak via Selection API'>
+<link rel='help' href='https://w3c.github.io/webcomponents/spec/shadow/'>
+<script src='/resources/testharness.js'></script>
+<script src='/resources/testharnessreport.js'></script>
+</head>
+<body>
+<div id='log'></div>
+<div id='container'></div>
+</body>
+<script>
+  'use strict';
+
+  const container = document.getElementById('container');
+
+  function reset(text) {
+    const host = container.appendChild(document.createElement('div'));
+    const shadow = host.attachShadow({mode: 'closed'});
+    const child = shadow.appendChild(document.createElement('span'));
+    child.textContent = text;
+    return shadow;
+  }
+
+  test(() => {
+    const text = 'text inside shadow';
+    const shadow = reset(text);
+    find(text);
```

Comment:
This test assumes that "find" will find contents within shadow DOM. That is not the case in Safari. I don't think this test can be merged without further discussion about this.